docs/migration.md

```diff
@@ -71,6 +75,47 @@ from pydantic.v1.utils import lenient_isinstance
 
 Pydantic V1 documentation is available at [https://docs.pydantic.dev/1.10/](https://docs.pydantic.dev/1.10/).
 
+
+#### Using Pydantic v1 features in a v1/v2 environment
+
+As of `pydantic>=1.10.17` the `pydantic.v1` namespace is available to use within v1,
+meaning that in order to unpin a `pydantic<2` dependency to allow using `v1` features,
+simply replace `pydantic<2` with `pydantic>=1.10.17,<3` and find and replace all:
+
+```python test="skip" lint="skip" upgrade="skip"
+from pydantic.<module> import <object>
+```
+
+to:
+
+```python test="skip" lint="skip" upgrade="skip"
+from pydantic.v1.<module> import <object>
+```
+
+Imports for `pydantic` v1 features for different versions of pydantic.
+
+=== "`pydantic>2,<3`"
+
+    ```python test="skip" lint="skip" upgrade="skip"
+    try:
+        from pydantic.v1.fields import ModelField
+    except ImportError:
+        from pydantic.fields import ModelField
+    ```
+
+=== "`pydantic>=1.10.17,<3`"
+
+    ```python test="skip" lint="skip" upgrade="skip"
+    from pydantic.v1.fields import ModelField
+    ```
+
+!!! note
+    When importing modules using `pydantic>=1.10.17,<2` with the `.v1` namespace
+    these modules will *not* be the same module as the same import without the `.v1`
+    namespace, but the symbols imported *will* be. For example ``pydantic.v1.fields is not pydantic.fields``
+    but ``pydantic.v1.fields.ModelField is pydantic.fields.ModelField``.
```

Comment:
Maybe add one more line about the fact that this is not something that's likely to be relevant for 99% of use cases, just a complication of our attempt to support a smoother upgrade experience?